doc/design/OrionCKANSink.md

```diff
@@ -206,10 +225,57 @@ A configuration example could be:
     cygnusagent.sinks.ckan-sink.attr_persistence = column
     cygnusagent.sinks.ckan-sink.orion_url = http://localhost:1026
 
-## Use cases
+[Top](#top)
+
+##<a name="section3"></a>Use cases
 Use `OrionCKANSink` if you are looking for a database storage not growing so much in the mid-long term.
 
-## Contact
+[Top](#top)
+
+##<a name="section4"></a>Implementation details
+###<a name="section4.1"></a>`OrionCKANSink` class
+As any other NGSI-like sink, `OrionCKANSink` extends the base `OrionSink`. The methods that are extended are:
+
+    void persist(Map<String, String>, NotifyContextRequest) throws Exception;
+    
+The context data, already parsed by `OrionSink` in `NotifyContextRequest`, is iterated and persisted in the CKAN backend by means of a `CKANBackendImpl` instance. Header information from the `Map<String, String>` is used to complete the persitence process, such as the timestamp or the destination.
+    
+    public void start();
+
+`CKANBackendImpl` is created. This must be done at the `start()` method and not in the constructor since the invoking sequence is `OrionCKANSink()` (contructor), `configure()` and `start()`.
+
+    public void configure(Context);
+    
+A complete configuration as the described above is read from the given `Context` instance.
+
+[Top](#top)
+
+###<a name="section4.2"></a>`CKANBackendImpl` class
+This is a convenience backend class for CKAN that extends the `HttpBackend` abstract class (provides common logic for any Http connection-based backend) and implements the `CKANBackend` interface (provides the methods that any CKAN backend must implement). Relevant methods are:
+
+    public void persist(long recvTimeTs, String recvTime, String orgName, String pkgName, String resName, String attrName, String attrType, String attrValue, String attrMd) throws Exception;
+    
+Persists the context data regarding a single entity's attribute within the datastore associated to the given resource. This resource belongs to the given package/dataset, which in the end belongs to the given organization as well. Other notified attributes will be persisted by using this method, next to current one. This method creates the parts of the hierarchy (organization-pacakge/dataset-resource-datastore) if any of them is missing (row-like mode).
```

Comment:
Fixed in https://github.com/telefonicaid/fiware-cygnus/commit/96c960ccecc8c9d05818bc84d0b15516f3c03253